File: solutions/scala/armstrong-numbers/1/src/main/scala/ArmstrongNumbers.scala
object ArmstrongNumbers {
  def isArmstrongNumber(n: Int): Boolean = {
    val dgs = digits(n)
    val armstrong = dgs.map { x =>
      scala.math.pow(x, dgs.size)
    }.sum
    n == armstrong
  }

  private def digits(n: Int): List[Int] = {
    var a = n
    val dgs = scala.collection.mutable.ListBuffer.empty[Int]
    while (a > 9) {
      val r = a % 10
      a /= 10
      dgs += r
    }
    dgs += a
    dgs.toList
  }
}

File: solutions/javascript/armstrong-numbers/1/armstrong-numbers.js
export const isArmstrongNumber = (n) => {
  const bigN = BigInt(n);
  const ds = digits(bigN);
  let armstrong = 0n;
  for(const x of ds) {
    armstrong += x ** BigInt(ds.length);
  }
  return armstrong === bigN;
};

function digits(n) {
  const ds = [];
  while(n > 9) {
    ds.push(n % 10n);
    n = n / 10n;
  }
  ds.push(n);
  return ds;
}